subsys/bluetooth/controller/hci_driver.c

```diff
@@ -447,6 +473,24 @@ static int hci_driver_open(void)
 		return -ENOMEM;
 	}
 
+	entropy_source = device_get_binding(DT_LABEL(DT_NODELABEL(rng)));
+	if (!entropy_source) {
+		BT_ERR("An entropy source is required");
+		return -EINVAL;
+	}
+
+	sdc_rand_source_t rand_functions = {
+		.rand_prio_low_get = rand_prio_low_vector_get,
+		.rand_prio_high_get = rand_prio_high_vector_get,
+		.rand_poll = rand_prio_low_vector_get_blocking
+	};
+
+	err = sdc_rand_source_register(&rand_functions);
+	if (err) {
+		BT_ERR("Failed to register randomness source with the SoftDevice Controller");
```

Comment:
Maybe add the err code returned to the log
```suggestion
		BT_ERR("Failed to register rand source (%d)", err);
```